server/src/main/java/org/opensearch/action/admin/cluster/decommission/awareness/put/DecommissionRequest.java

```diff
@@ -96,6 +107,42 @@ public boolean isNoDelay() {
         return noDelay;
     }
 
+    /**
+     * Sets retryOnClusterManagerChange for decommission request
+     *
+     * @param retryOnClusterManagerSwitch boolean for request to retry decommission action on cluster manager switch
+     * @return this request
+     */
+    public DecommissionRequest setRetryOnClusterManagerSwitch(boolean retryOnClusterManagerSwitch) {
+        this.retryOnClusterManagerSwitch = retryOnClusterManagerSwitch;
+        return this;
+    }
+
+    /**
+     * @return Returns whether decommission is retry eligible on cluster manager switch
+     */
+    public boolean retryOnClusterManagerSwitch() {
+        return this.retryOnClusterManagerSwitch;
+    }
+
+    /**
+     * Sets the timeout for the request
+     *
+     * @param timeout time out for the request
+     * @return this request
+     */
+    public DecommissionRequest setTimeout(TimeValue timeout) {
+        this.timeout = timeout;
+        return this;
+    }
+
+    /**
+     * @return timeout
+     */
+    public TimeValue timeout() {
+        return this.timeout;
+    }
```

Comment:
What is the upper bound for timeout?

Reply:
Didn't kept a upper bound. Although should be atleast 2 min